doc/en/user/source/community/oauth2/index.rst

```diff
@@ -0,0 +1,151 @@
+.. _security_tutorials_oauth2:
+
+Authentication with OAuth2
+==========================
+
+This tutorial introduces GeoServer OAuth2 support and walks through the process of
+setting up authentication aganist an OAuth2 provider. It is recommended that the 
+:ref:`security_auth_chain` section be read before proceeding.
+
+OAuth2 Protocol and GeoServer OAuth2 core module
+------------------------------------------------
+
+This module allows GeoServer to authenticate against the `OAuth2 Protocol <https://tools.ietf.org/html/rfc6749>`_.
+
+In order to let the module work, it's mandatory to setup and configure both ``oauth2`` and ``oauth2-xxxx-extension``.
+
+The first one contains the necessary dependencies of the OAuth2 core module. This module contains the implementation of the 
+GeoServer security filter, the base classes for the OAuth2 Token services and the GeoServer GUI panel.
+
+Since in almost all cases the only thing different between OAuth2 Providers are the endpoint URIs and the client connection
+information (not only the keys - public and secret - but also the user profile representations), in order to allow GeoServer
+connecting to a specific OAuth2 provider it is sufficient to install the OAuth2 Core module plugin (and correctly configure
+the parameters through the GeoServer GUI - see next section for the details) and the concrete implementation of the OAuth2
+REST token template and resource details.
+
+Currently this module is shipped with a sample extension for Google OAuth2 Provider. This is a particular case since the 
+Google JWT response is not standard and therefore we had to define and inject also a ``GoogleUserAuthenticationConverter`` taking
+the Google REST response against a valid ``access_token`` and converting it to a OAuth2 standard one.
+
+Other than this the most interesting part is the implementation of the base class ``GeoServerOAuth2SecurityConfiguration``.
+
+The latter contains the Google implementation of the ``OAuth2RestTemplate``.
+
+In the next section we will see how to install and configure the OAuth2 security filter on GeoServer authenticating against 
+Google OAuth2 Provider.
+
+Configure the Google authentication provider
+--------------------------------------------
+
+The first thing to do is to configure the OAuth2 Provider and obtain ``Client ID`` and ``Client Sectet`` keys.
+
+#. Obtain OAuth 2.0 credentials from the Google API Console.
+
+   Visit the `Google API Console <https://console.developers.google.com/>`_ to obtain OAuth 2.0 credentials such as a client ID and client secret 
+   that are known to both Google and your application. The set of values varies based on what type of application you are building. 
+   For example, a JavaScript application does not require a secret, but a web server application does.
+   
+    .. figure:: images/google1.jpg
+       :align: center
+
+#. Create a new ``OAuth client ID``
```

Comment:
Looks like there is a missing step here, when I get to the credentials page I'm offered to create a new project:

![selezione_062](https://cloud.githubusercontent.com/assets/325433/18394831/298e01ea-76bc-11e6-86f9-048ef4ffe88a.png)

and then:

![selezione_064](https://cloud.githubusercontent.com/assets/325433/18394875/4bb7217a-76bc-11e6-8f63-e2b86ff65384.png)

If the project already exists then I am offered to select it. Only after that I reach the screen at point 2.